### PR TITLE
AOV cleanups

### PR DESCRIPTION
**Pull request by est77**
- Fix AOV entity names.
- Removed Frame add / transfer AOVs. Frame is now inmutable again.
- Do not reorder AOVs when saving projects. This fixes parts order in multi-part EXR images.

### CODE REVIEW

File: src/appleseed/renderer/modeling/frame/frame.cpp [hunk 1]
```diff
@@ -131,6 +134,35 @@ Frame::Frame(
...
+        if (aov_index == size_t(~0) && aov_images().size() < MaxAOVCount)
+        {
+            const IAOVFactory* aov_factory = aov_registrar.lookup(original_aov->get_model());
+            assert(aov_factory);
+
+            auto_release_ptr<AOV> aov = aov_factory->create(original_aov->get_parameters());
+
+            aov_images().append(
+                aov->get_name(),
+                4, // aov->get_channel_count(),
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/frame/frame.cpp]: Why is 4 hardcoded here?

File: src/appleseed/renderer/modeling/frame/frame.cpp [hunk 2]
```diff
@@ -131,6 +134,35 @@ Frame::Frame(
...
+                PixelFormatFloat);
+            impl->m_aovs.insert(aov);
+        }
+        else
+        {
+            RENDERER_LOG_WARNING(
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/frame/frame.cpp]: Now that you copy all AOVs at once in a loop, you can check before hand whether they'll all fit or not, and emit a warning message only once.

File: src/appleseed/renderer/modeling/project/projectfilereader.cpp [hunk 3]
```diff
@@ -2405,18 +2405,78 @@ namespace
...
+                    m_registrar.lookup(m_model.c_str());
+
+                if (factory)
+                {
+                    return factory->create(m_params);
+                }
+                else
+                {
+                    RENDERER_LOG_ERROR(
+                        "while defining AOV: invalid model \"%s\".",
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/project/projectfilereader.cpp]: *aov* (lower case) in log messages.

File: src/appleseed/renderer/modeling/frame/frame.cpp (hunk 1 above)
Comment (est77) [reply src/appleseed/renderer/modeling/frame/frame.cpp]: It comes from the original code. I added the comment to not forget about this.
I didn't check yet if there's any issue passing the correct number of channels here.

File: src/appleseed/renderer/modeling/frame/frame.cpp (hunk 2 above)
Comment (est77) [reply src/appleseed/renderer/modeling/frame/frame.cpp]: Good point

File: src/appleseed/renderer/modeling/project/projectfilereader.cpp (hunk 3 above)
Comment (est77) [reply src/appleseed/renderer/modeling/project/projectfilereader.cpp]: Fixed and rebased

**Comment by est77** [comment]
Fixed and rebased

File: src/appleseed/renderer/modeling/project/projectfilereader.cpp [hunk 4]
```diff
@@ -2405,18 +2405,78 @@ namespace
...
+                        m_model.c_str());
+                    m_context.get_event_counters().signal_error();
+                }
+            }
+            catch (const ExceptionDictionaryKeyNotFound& e)
+            {
+                RENDERER_LOG_ERROR(
+                    "while defining AOV \"%s\": required parameter \"%s\" missing.",
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/project/projectfilereader.cpp]: *aov*

File: src/appleseed/renderer/modeling/project/projectfilereader.cpp [hunk 5]
```diff
@@ -2405,18 +2405,78 @@ namespace
...
+                    m_model.c_str(),
+                    e.string());
+                m_context.get_event_counters().signal_error();
+            }
+            catch (const ExceptionUnknownEntity& e)
+            {
+                RENDERER_LOG_ERROR(
+                    "while defining AOV \"%s\": unknown entity \"%s\".",
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/project/projectfilereader.cpp]: *aov*

**Pull request merged**